File: api/Models/Collateral.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using api.Enums;

namespace api.Models;

public class Collateral
{
    [Key]
    public int Id { get; set; }
    [Required]
    public string? Title { get; set; }
    public string? Description { get; set; }
    [Required]
    public CollateralAgreementType AgreementType { get; set; }
    public decimal Value { get; set; }
    [Required]
    public CollateralCondition Condition { get; set; } = CollateralCondition.Stable;
    [Required]
    public CollateralStatus Status { get; set; } = CollateralStatus.Active;
    public string? DocumentUrl { get; set; }
    [Required]
    public string? OwnerId { get; set; }
    [Required]
    public int LoanId { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.Now;
    public DateOnly? ExpirationDate { get; set; }
    public string? Location { get; set; }

    // Navigation Properties
    [ForeignKey("OwnerId")]
    public AppUser? Owner { get; set; }
    [ForeignKey("LoanId")]
    public Loan? Loan { get; set; }
    public List<Photo> Photos { get; set; } = [];
    public List<FileUpload> Files { get; set; } = [];
}

File: frontend/src/models/collateral.ts
import { ApiFile } from "./apiFile";
import { CollateralAgreementType } from "./collateralAgreementType";
import { CollateralCondition } from "./collateralCondition";
import { CollateralStatus } from "./collateralStatus";
import { Loan } from "./loan";
import { Photo } from "./photo";
import { User } from "./user";

export interface Collateral {
  id: number;
  title: string;
  photo: Photo;
  description: string;
  value: number;
  documentUrl: string;
  ownerId: string;
  condition: CollateralCondition;
  createdAt: string;
  status: CollateralStatus;
  owner: User;
  loanId: number;
  loan: Loan;
  photos: Photo[];
  agreementType: CollateralAgreementType;
  location?: string;
  expirationDate?: string;
  files: ApiFile[];
}
